From bbfd56ba3d2c2dfed90773a62f857d152d3767cd Mon Sep 17 00:00:00 2001
From: Sunil Kumar Kori <sunil.kori@nxp.com>
Date: Thu, 22 Feb 2018 15:06:46 +0530
Subject: [PATCH 157/166] README: Update for l3fwd with eventdev config

README document is updated with newly added eventdev support for:
 -- parallel queue
 -- atomic queue

Signed-off-by: Sunil Kumar Kori <sunil.kori@nxp.com>
---
 nxp/README | 46 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/nxp/README b/nxp/README
index d86177de0ecb..76c44742b448 100644
--- a/nxp/README
+++ b/nxp/README
@@ -405,6 +405,52 @@ Following are some pre-requisites on either platform.
       For traffic to port 4: 4.1.1.0/24
 
 
+  Layer-3 Forwarding 'l3fwd' Application using eventdev
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  Execute following commands to run 'l3fwd' application with eventdev
+
+  Note:
+    1. For DPAA1 platforms
+
+	-> disable push mode queue, using 'export DPAA_PUSH_QUEUES_NUMBER=0'
+
+	-> virtual device should be "event_dpaa1" i.e. --vdev="event_dpaa1"
+
+    2. For DPAA2 platforms
+
+	-> virtual device should be "event_dpaa2" i.e. --vdev="event_dpaa2"
+
+    3. Only single instance of virtual device(vdev) is allowed.
+
+  -- With parallel queue configuration
+
+  For 1 core, 1 Port, 1 queue per port, 1 eventdev, 1 event queue, 1 event port =>
+
+  $ l3fwd -c 0x08 -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,3)" -e="(0,1,1)" -a="(5,0,2,0,0,0)" -l="(0,0,0,3)" -P
+
+  For 2 core, 1 Port, 2 queue per port, 1 eventdev, 1 event queue, 2 event port =>
+
+  $ l3fwd -c 0x0C -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,2),(5,1,3)" -e="(0,1,2)" -a="(5,0,2,0,0,0),(5,1,2,0,0,0)" -l="(0,0,0,2),(1,0,0,3)" -P
+
+  For 4 core, 1 Port, 4 queue per port, 1 eventdev, 1 event queue, 4 event port =>
+
+  $ l3fwd -c 0x0F -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,0),(5,1,1),(5,2,2),(5,3,3)" -e="(0,1,4)" -a="(5,0,2,0,0,0),(5,1,2,0,0,0),(5,2,2,0,0,0),(5,3,2,0,0,0)" -l="(0,0,0,0),(1,0,0,1),(2,0,0,2),(3,0,0,3)" -P
+
+  -- With atomic queue configuration
+
+  For 1 core, 1 Port, 1 queue per port, 1 eventdev, 1 event queue, 1 event port =>
+
+  $ l3fwd -c 0x08 -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,3)" -e="(0,1,1)" -a="(5,0,1,0,0,0)" -l="(0,0,0,3)" -P
+
+  For 2 core, 1 Port, 2 queue per port, 1 eventdev, 1 event queue, 2 event port =>
+
+  $ l3fwd -c 0x0C -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,2),(5,1,3)" -e="(0,1,2)" -a="(5,0,1,0,0,0),(5,1,1,0,0,0)" -l="(0,0,0,2),(1,0,0,3)" -P
+
+  For 4 core, 1 Port, 4 queue per port, 1 eventdev, 1 event queue, 4 event port =>
+
+  $ l3fwd -c 0x0F -n 1 --vdev="event_dpaa1" -- -p 0x20 --config="(5,0,0),(5,1,1),(5,2,2),(5,3,3)" -e="(0,1,4)" -a="(5,0,1,0,0,0),(5,1,1,0,0,0),(5,2,1,0,0,0),(5,3,1,0,0,0)" -l="(0,0,0,0),(1,0,0,1),(2,0,0,2),(3,0,0,3)" -P
+
   Running DPDK Applications in Virtual Machine
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-- 
2.14.3

